tests/unit/test_application.py

```diff
@@ -1355,6 +1356,11 @@ def grammar_project_mini(tmp_path):
           - on-amd64-to-riscv64
         - on amd64 to s390x:
           - on-amd64-to-s390x
+
+        # non-grammar data in a non-grammar keyword
```

Comment:
can you check that `meson-parameters` stays as `['foo', 'bar']` in one of the tests (or a new one)?